### PR TITLE
feat: add Easy LangServe Apps with Pulumi on AWS blog

### PR DESCRIPTION
**Pull request by dirien**
## Checklist:

- [x] I have reviewed the [style guide](https://github.com/pulumi/pulumi-hugo/blob/master/STYLE-GUIDE.md).
- [x] If blogging, I have reviewed the [blogging guide](https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md).
- [x] I have manually confirmed that all new links work.
- [x] I added aliases (i.e., redirects) for all filename changes.
- [x] If making css changes, I rebuilt the bundle.

### CODE REVIEW

File: themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+authors:
+- engin-diri
+
+tags:
+- ai
+- langserve
+- aws
+---
+
+We all know, how easy it is to create, deploy, and manage any cloud infrastructure with Pulumi using your favorite
```
Comment (ericrudder) [inline themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md:15]:
```suggestion
We all know how easy it is to create, deploy, and manage any cloud infrastructure with Pulumi using your favorite
```

File: themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+
+```shell
+pulumi destroy
+```
+
+All the resources will be deleted, including the Pinecone index.
+
+## In Conclusion
+
+In this blog post, we showed you how easy it is to create, deploy, and manage LangServe apps with Pulumi. Using the
```
Comment (ericrudder) [inline themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md:397]:
```suggestion
In this blog post, we showed you how easy it is to create, deploy, and manage LangServe apps with Pulumi. The
```

File: themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+example projects from the Pulumi examples repository, is a creating starting point to build your own LLM-powered apps.
```
Comment (ericrudder) [inline themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md:398]:
```suggestion
projects from the Pulumi examples repository provide an excellent starting point to build your own LLM-powered apps.
```

File: themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md [hunk 4]
```diff
@@ -0,0 +1,440 @@
...
+
+tags:
+- ai
+- langserve
+- aws
+---
+
+![Easy LangServe Apps with Pulumi on AWS](./meta.png)
+
+We all know, how easy it is to create, deploy, and manage any cloud infrastructure
```
Comment (scottslowe) [inline themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md:17]:
```suggestion
We all know how easy it is to create, deploy, and manage any cloud infrastructure
```

File: themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md [hunk 5]
```diff
@@ -0,0 +1,440 @@
...
+with [Pulumi](https://www.pulumi.com/) using your favorite
+programming language. With the rise of artificial intelligence (AI) more and more developers are working on[LLM-powered
```
Comment (scottslowe) [inline themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md:19]:
```suggestion
programming language. With the rise of artificial intelligence (AI) more and more developers are working on [LLM-powered
```

File: themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md [hunk 6]
```diff
@@ -0,0 +1,440 @@
...
+applications and services](https://www.thoughtworks.com/en-de/radar/techniques/llm-powered-autonomous-agents). And with
+this, the need for have the same ease of use for creating, deploying, and managing
```
Comment (scottslowe) [inline themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md:21]:
```suggestion
this, the need to have the same ease of use for creating, deploying, and managing
```

File: themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md [hunk 7]
```diff
@@ -0,0 +1,440 @@
...
+
+In this blog post, we will show you how to this can be achieved with combining Pulumi
+and [LangServe](https://python.langchain.com/docs/langserve).
+
+## What is 🦜️🏓 LangServe?
+
+[LangServe](https://github.com/langchain-ai/langserve) is part of the
+awesome [🦜️🔗 LangChain](https://github.com/langchain-ai/langchain)
+framework. LangChain simplifies the process of creating generative AI application interfaces. Generative AI applications
+are powered by large language models (LLMs) and deliver an answer to your question (so-called prompts) in natural
```
Comment (scottslowe) [inline themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md:32]:
```suggestion
are powered by large language models (LLMs) and deliver an answer to your question (called prompts) in natural
```

File: themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md [hunk 8]
```diff
@@ -0,0 +1,440 @@
...
+popular use cases could be:
+
+- Ask a question over your company's internal knowledge base
+- Summarize a document
+- Text extraction or data labeling
+
+Building these types of integrations often needs to build pipelines (called typically chains), starting from the prompt
+and enriching the prompt with your own data. Here is where LangChain comes into play.
+
+LangChain gives you all the abstractions you need to start building your own LLM app. It has many so-called
```
Comment (scottslowe) [inline themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md:43]:
```suggestion
LangChain gives you all the abstractions you need to start building your own LLM app. It has many
```

File: themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md [hunk 9]
```diff
@@ -0,0 +1,440 @@
...
+are [available here](https://github.com/pulumi/examples/) and ready to
+use as a base for our two demo projects.
+
+### Create a Gandalf Chatbot App 🧙‍♂️
+
+{{% notes type="info" %}}
+Please make sure that you have `OPENAI_API_KEY` environment variable set with your OpenAI API key.
+{{% /notes %}}
+
+Our first example, is a chatbot app with a twist. We will use LangChain to create chain that will answer questions
```
Comment (scottslowe) [inline themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md:79]:
```suggestion
Our first example is a chatbot app with a twist. We will use LangChain to create chain that will answer questions
```

File: themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md [hunk 10]
```diff
@@ -0,0 +1,440 @@
...
+
+{{% notes type="info" %}}
+Please make sure that you have the following API keys set:
+
+- `PINECONE_API_KEY` environment variable set with your Pinecone API key. You can get it from the Pinecone console.
+- `OPENAI_API_KEY` environment variable set with your OpenAI API key.
+
+{{% /notes %}}
+
+Our second example, will show you how to create a RAG (Retrieval-Augmented Generation) app
```
Comment (scottslowe) [inline themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md:197]:
```suggestion
Our second example will show you how to create a RAG (Retrieval-Augmented Generation) app
```

File: themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md [hunk 11]
```diff
@@ -0,0 +1,440 @@
...
+with [Pinecone](https://pinecone.io/) vector
+databases. The clue of this example here is to show you, how easy you can extend the Pulumi LangServe example to deploy
```
Comment (scottslowe) [inline themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md:199]:
```suggestion
databases. The clue of this example here is to show you how easy you can extend the Pulumi LangServe example to deploy
```

File: themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md [hunk 12]
```diff
@@ -0,0 +1,440 @@
...
+```python
+# ... omitted code
+{
+    "name": "PINECONE_API_KEY",
+    "valueFrom": langserve_ssm_parameter_pinecone_name
+}
+# ... omitted code
+```
+
+This should create your Pinecone index and update the Fargate task definition to use the Pinecone API key, what we are
```
Comment (scottslowe) [inline themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md:298]:
```suggestion
This should create your Pinecone index and update the Fargate task definition to use the Pinecone API key; this is what we are
```

File: themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md [hunk 13]
```diff
@@ -0,0 +1,440 @@
...
+going to need to use the Pinecone index in our LangServe app.
```
Comment (scottslowe) [inline themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md:299]:
```suggestion
going to need in order to use the Pinecone index in our LangServe app.
```

File: themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md [hunk 14]
```diff
@@ -0,0 +1,440 @@
...
+
+To use the Pinecond index in our LangServe app, we need first to add following packages to our LangChain app:
```
Comment (scottslowe) [inline themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md:301]:
```suggestion
To use the Pinecone index in our LangServe app, we need first to add following packages to our LangChain app:
```

File: themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md [hunk 15]
```diff
@@ -0,0 +1,440 @@
...
+Here a screenshot of the Pinecone console:
+
+![Pinecone Console](./pinecone-console.png)
+
+And if you want to test the LangServe app, you can use the playground UI, where you should be able to ask questions
+about Gandalf the Grey and get answers based on the documents ingested into the Pinecone index.
+
+![LangServe Pinecone](./langserve-pinecone.png)
+
+If you want to destroy the stack, you can use the following command:
```
Comment (scottslowe) [inline themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md:415]:
```suggestion
When you're ready to destroy the stack, you can use the following command:
```

File: themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md [hunk 16]
```diff
@@ -0,0 +1,440 @@
...
+
+```shell
+pulumi destroy
+```
+
+The Pulumi CLI will ask you to confirm the changes, and if you are happy with the changes, you can confirm them by
+typing `yes` and all the resources will be deleted, including the Pinecone index.
+
+All the resources will be deleted, including the Pinecone index.
```
Comment (scottslowe) [inline themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md:424]: Seems redundant with the previous paragraph?

**Review by scottslowe — CHANGES_REQUESTED** [review]
Numerous minor/small grammatical corrections and updates, but the core of the blog post is really good.

File: themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md (hunk 16 above)
Comment (dirien) [reply themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md:424]: so true! 😅

File: themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md [hunk 17]
```diff
@@ -0,0 +1,437 @@
...
+authors:
+- engin-diri
+
+tags:
+- ai
+- langserve
+- aws
+---
+
+![Easy LangServe Apps with Pulumi on AWS](./meta.png)
```
Comment (cnunciato) [inline themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md:15]: For future, please don't add these images here like this. If we want to have these images appear at the top of a blog post, we can do it for all posts in the template.

Comment (dirien) [reply themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md:15]: @cnunciato should I remove it?

Comment (cnunciato) [reply themes/default/content/blog/easy-langserve-apps-with-pulumi/index.md:15]: All good, I'll fix it up later -- going to add a timestamp to the post in a moment to make sure it sorts correctly once Luke's post goes out.

**Pull request merged**